### PR TITLE
CB-365 new ImageCatalog tests for get images for an existing stack

### PR DESCRIPTION
**Pull request by biharitomi**
An issue was discovered during the new test implementations and that test could be enabled again if we will have finished the fix.

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/ImageCatalogTest.java [hunk 1]
```diff
@@ -398,5 +402,76 @@ public void testGetListOfImageCatalogs(TestContext testContext) {
...
+    public void testGetImageCatalogByNameAndStackWhenPreWarmedImageHasBeenUsed(TestContext testContext) {
+        createDefaultCredential(testContext);
+        initializeDefaultClusterDefinitions(testContext);
+
+        String imgCatalogName = getNameGenerator().getRandomNameForResource();
+        String stackName = imgCatalogName + "-stack";
+        MockedTestContext mockedTestContext = (MockedTestContext) testContext;
+
+        testContext
+                .given(ImageCatalogDto.class).withName(imgCatalogName).withUrl(mockedTestContext.getImageCatalogMockServerSetup().getPreWarmedImageCatalogUrl())
```
Comment (attilapalfi92) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/ImageCatalogTest.java]: 'ImageCatalogDto' should be named as ImageCatalogTestDto

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/action/imagecatalog/ImageCatalogGetImagesByNameAction.java [hunk 2]
```diff
@@ -17,29 +20,47 @@
...
         } catch (Exception e) {
             LOGGER.warn("Cannot get images of ImageCatalog : {}", entity.getRequest().getName());
             throw e;
         }
         return entity;
     }
+
+    private ImagesV4Response getImagesV4Response(ImageCatalogDto entity, CloudbreakClient cloudbreakClient, ImageCatalogV4Endpoint imageCatalogV4Endpoint)
+            throws Exception {
+        ImagesV4Response imagesByName;
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/action/imagecatalog/ImageCatalogGetImagesByNameAction.java]: Personally i prefer
```
return StringUtils.isNoneEmpty(stackName)
            ? imageCatalogV4Endpoint
                     .getImagesByName(cloudbreakClient.getWorkspaceId(), entity.getName(), stackName, null)
            : imageCatalogV4Endpoint
                     .getImagesByName(cloudbreakClient.getWorkspaceId(), entity.getName(), null, platform.name());
```

Comment (biharitomi) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/action/imagecatalog/ImageCatalogGetImagesByNameAction.java]: Thanks, it looks better this time, will change it if stg needs to be modified around there. :)

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/ImageCatalogTest.java (hunk 1 above)
Comment (biharitomi) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/ImageCatalogTest.java]: :D Haha it's too late, but will rename it if there will be modifications around there.

Comment (attilapalfi92) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/ImageCatalogTest.java]: no need for it, I've done it my branch

Comment (biharitomi) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/ImageCatalogTest.java]: thanks

**Pull request merged**